andela/the_spinners-frontend
# #169817490 Integrate Notification Pane

#### What does this PR do? 
```
Integrate Notification pane
```
#### Description of Task to be completed?
- Integrate notification pane
- Allow user to set preferences
- User can get real-time notifications with socket.io
- Refactor the APP LAYOUT COMPONENTS(Header, Sidebar, Footer) to be  dynamic. not requiring devs to copy it to their pages
- enable profile picture to be dynamic on profile change
#### How should this be manually tested?
- On the menu notification icon, there should be unread notification if any on load. Any other notification will also be there on new requests,
- Create a new request on the backend API with your credentials. then your manager should get a notification. A request approved, then you should get notification.
 example of request: [https://snippers-backend-stage.herokuapp.com/docs/#/trips/post_api_trips_one_way](https://snippers-backend-stage.herokuapp.com/docs/#/trips/post_api_trips_one_way)
#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[ #169817490](https://www.pivotaltracker.com/story/show/169817490) Implement notifications pane 
[#169817505](https://www.pivotaltracker.com/story/show/169817505) Marking all notifications as read
[#169817503](https://www.pivotaltracker.com/story/show/169817503) Users should see a toast for new notifications
#### Screenshots (if appropriate)
<img width="387" alt="Screen Shot 2020-02-19 at 11 15 25" src="https://user-images.githubusercontent.com/52409462/74819510-2cd21d00-5309-11ea-8cbc-5896895f647a.png">

#### Questions:
